### ISSUE TITLE
Provider Versioning

### ISSUE THREAD

**Issue by yoimnik** [CityOfLosAngeles/mobility-data-specification#313]
If a client requests a version that is not supported, the spec says:
```
If an unsupported or invalid version is requested, the API must respond with a status of 406 Not Acceptable. In which case, the response should include a body specifying a list of supported versions.
```
can we go into more detail for what this body would look like? a typical 406 response would contain a `Vary` header with values as a comma separated list of header _types_ rather than header values (ie. `{ 'Vary': 'Accept' }`. is there a spec for what the response body should look like for the version values?

**Comment by stephanBsg** [comment]
Hello,
If it's okay, I'd like to share with you my past experience on the matter
of API versions.
I've been programming API for years and versions is always a challenge as
APIs evolve rapidly but the clients don't, often, because it requires code
changes, re-compile, resources and publishing.
When an API is used by multiple tiers and the API evolves into a not
backwards compatible version, client applications can break.
An easy way to resolve this is to use routes and to point urls to a
specific path bound to it's version: v2/my_api_endpoint/,
v2.2/my_api_endpoint/, v3/my_api_endpoint, etc.
There's pros and cons to this:
Pros:
- Client applications don't break when the API changes.
- Client apps can use a simpler version when they don't need all the new
features of newer API versions
- Tiers partners have more time to make their changes and comply to the
last API version requirements
- API Requests are easier to build as the version is passed through the URL
rather than the headers.

Cons:
- API data sources might not be compatible with multiple versions (example:
database rdms)
- Multiple API versions are more work to maintain if they are all fully
supported.
- New rules that needs immediate compliance can be only enforced through
shutting down older versions or by updating all versions.

I hope this helped.
Thank you for your time and future feedbacks.

Steph.

**Comment by billdirks** [comment]
This issue was discussed in the 6/6/2019 MDS specification meeting. We came to consensus that, since there already a specified way to negotiate versions via a OPTIONS method, we should use that instead of specifying 2 ways to get version information. So the proposal put forth was, if a 406 is returned from the server, the client should use the OPTIONS method to negotiate versions and the response body of the 406 would not need to have list version numbers in the response body.

Let me know if you have any concerns with this resolution @yoimnik

**Comment by hunterowens** [comment]
fixed in #327
